### ISSUE TITLE
[Dancer-1.3203-TRIAL] empty port not found

### ISSUE THREAD

**Issue by eserte** [PerlDancer/Dancer#1183]
On my FreeBSD smokers the t/02_request/20_body.t fails:
```
empty port not found at /usr/perl5.16.3t/lib/site_perl/5.16.3/Net/EmptyPort.pm line 64.
# Looks like your test exited with 49 before it could output anything.
t/02_request/20_body.t .............................. 
Dubious, test returned 49 (wstat 12544, 0x3100)
Failed 2/2 subtests 
```

**Comment by bigpresh** [comment]
Hmm.  I think this'll be because of PR #1150 - where I had it listen on 127.0.0.10 instead of 127.0.0.1, to avoid an annoying race condition where the port that Net::EmptyPort selects is taken by something else between our test picking it and using it - but that solution doesn't work on FreeBSD, where the loopback interface only has 127.0.0.1/32 by default, not 127/8 like Linux, Windows and some others.

So, I need to either revert that (and, in the process, bring back the race condition), or find a better way.

OTTOMH, one way would be to attempt to bind to port 0 on 127.0.0.10 (or other random IP in 127/8) - if it works, great, use that port, if not, try again on 127.0.0.1 and accept the potential race condition.  I don't really like changing the test behaviour depending on the platform in use, but it may be necessary.

**Comment by bigpresh** [comment]
*Hopefully* this issue will have been resolved by my PR #1185 - the reports for Dancer-1.3204-TRIAL on cpantesters show plenty of passes from FreeBSD smokers, so I'm going to take that as a sign that the fix worked.  Not looking so good for Mac OSX, though - some odd failures there where the request hit a Squid proxy - hopefully that's a smoker misconfiguration but I'm not sure.
